### PR TITLE
Improve non immutable rendering

### PR DESCRIPTION
**Pull request by nhannah**
Fixes most of the issues related to remove immutable flat list renders. The next step is to break messages out to further improve the render cycle.

### CODE REVIEW

File: src/contexts/messageContext/MessageContext.tsx [hunk 1]
```diff
@@ -112,7 +113,10 @@ export type MessageContextValue<
   preventPress?: boolean;
   /** Whether or not the avatar show show next to Message */
   showAvatar?: boolean;
-};
+} & Pick<
+  ChannelContextValue<At, Ch, Co, Ev, Me, Re, Us>,
+  'channel' | 'readEventsEnabled' | 'disabled' | 'members'
```
Comment (virdesai) [inline src/contexts/messageContext/MessageContext.tsx:118]:
```suggestion
  'channel' | 'disabled' | 'members' | 'readEventsEnabled'
```

File: src/components/Message/hooks/useCreateMessageContext.ts [hunk 2]
```diff
@@ -62,6 +66,7 @@ export const useCreateMessageContext = <
   }${message.updated_at}${message.deleted_at}${readBy}${message.status}${
     message.type
   }${message.text}${message.reply_count}`;
+  const membersValue = Object.keys(members).join;
```
Comment (virdesai) [inline src/components/Message/hooks/useCreateMessageContext.ts:69]: ?
```suggestion
  const membersValue = Object.keys(members).join();
```

File: src/components/Message/MessageSimple/MessageStatus.tsx [hunk 3]
```diff
@@ -51,9 +47,8 @@ export type MessageStatusPropsWithContext<
   Us extends DefaultUserType = DefaultUserType
 > = Pick<
   MessageContextValue<At, Ch, Co, Ev, Me, Re, Us>,
-  'message' | 'threadList'
-> &
-  Pick<ChannelContextValue<At, Ch, Co, Ev, Me, Re, Us>, 'readEventsEnabled'>;
+  'message' | 'threadList' | 'readEventsEnabled'
```
Comment (virdesai) [inline src/components/Message/MessageSimple/MessageStatus.tsx:50]:
```suggestion
  'message' | 'readEventsEnabled' | 'threadList'
```

**Review by virdesai — APPROVED** [review]

File: src/components/Message/hooks/useCreateMessageContext.ts (hunk 2 above)
Comment (nhannah) [reply src/components/Message/hooks/useCreateMessageContext.ts:69]: 3 am stuff, leme make sure it doesn't change anything

File: src/components/Message/Message.tsx [hunk 4]
```diff
@@ -1048,9 +1057,11 @@ const MessageWithContext = <
                 targetedStyle,
               ]}
             />
-            <MessageProvider value={messageContext}>
-              <MessageSimple />
-            </MessageProvider>
+            <MessagesProvider value={messagesContext}>
```
Comment (vishalnarkhede) [inline src/components/Message/Message.tsx:1060]: Lets add some nice comment here for future purpose, why we have `MessagesProvider` here!!

**Review by vishalnarkhede — APPROVED** [review]

File: src/components/Message/Message.tsx (hunk 4 above)
Comment (nhannah) [reply src/components/Message/Message.tsx:1060]: Can do.

**Pull request merged**